library/std/src/sys/unix/thread.rs

```diff
@@ -84,6 +96,13 @@ impl Thread {
             };
         }
 
+        #[cfg(target_os = "horizon")]
+        {
+            // Set the priority and affinity values
+            assert_eq!(libc::pthread_attr_setpriority(&mut attr, priority), 0);
+            assert_eq!(libc::pthread_attr_setaffinity(&mut attr, affinity), 0);
```

Comment:
Ok, if it matches other calls to `libc::pthread_*` then consistency is probably best. I don't think we expect these calls to typically fail anyway...

Reply:
It would be a programming bug if the affinity is wrong, and probably same for priority, so I'll leave it as an assert (esp. since most of the other pthread calls do this, probably for the same reasons).